File: generated-code/codecraft/go/model/player.go
package model

import "io"
import . "trans_gen_test/stream"
import "fmt"

// Player (strategy, client)
type Player struct {
    // Player's ID
    Id int32
    // Current score
    Score int32
    // Current amount of resource
    Resource int32
}

func NewPlayer(id int32, score int32, resource int32) Player {
    return Player {
        Id: id,
        Score: score,
        Resource: resource,
    }
}

// Read Player from reader
func ReadPlayer(reader io.Reader) Player {
    var id int32
    id = ReadInt32(reader)
    var score int32
    score = ReadInt32(reader)
    var resource int32
    resource = ReadInt32(reader)
    return Player {
        Id: id,
        Score: score,
        Resource: resource,
    }
}

// Write Player to writer
func (player Player) Write(writer io.Writer) {
    id := player.Id
    WriteInt32(writer, id)
    score := player.Score
    WriteInt32(writer, score)
    resource := player.Resource
    WriteInt32(writer, resource)
}

// Get string representation of Player
func (player Player) String() string {
    stringResult := "{ "
    stringResult += "Id: "
    id := player.Id
    stringResult += fmt.Sprint(id)
    stringResult += ", "
    stringResult += "Score: "
    score := player.Score
    stringResult += fmt.Sprint(score)
    stringResult += ", "
    stringResult += "Resource: "
    resource := player.Resource
    stringResult += fmt.Sprint(resource)
    stringResult += " }"
    return stringResult
}

File: generated-code/codecraft/haskell/FileReadWrite/src/Model/Player.hs
module Model.Player where

import Prelude hiding (id)
import qualified Trans
import Trans (Trans)
import Data.Int

data Player = Player {
    id :: Int32,
    score :: Int32,
    resource :: Int32 }
    deriving Show

instance Trans Player where
    read = do
        id <- Trans.read
        score <- Trans.read
        resource <- Trans.read
        return Player {
            id,
            score,
            resource }

    write Player {
        id,
        score,
        resource } = do
            Trans.write id
            Trans.write score
            Trans.write resource